lucet-runtime/lucet-runtime-tests/src/entrypoint.rs

```diff
@@ -728,8 +852,11 @@ macro_rules! entrypoint_tests {
                 let func = vmctx
                     .get_func_from_idx(0, cb_idx)
                     .expect("can get function by index");
-                let func = func as *const extern "C" fn(*mut lucet_vmctx, u64) -> u64;
```

Comment:
This specific test was very handy because in switching to `FunctionHandle`, which has an `extern "C" fn()` rather than a `*const extern "C" fn()`. In following the types literally, I tried doing something like `&func.ptr as *const extern "C" fn...` and ran into the segfaults I mentioned earlier, with a pointer that had one too many indirections.

It turns out that most of our `*const extern "C" fn()` are _really_ `extern "C" fn()` -- they're function pointers, not pointers to function pointers. Because it's *much* nicer syntactically to work with (you can `as` them without issue), I'm actually not sure if that's intentional... it results in highly confusing errors when following the types literally though.

The last two commits push through a change to turn `*const extern "C" fn()` into `extern "C" fn()` where that's the actual pointer type of the underlying data, but I'm not sold on them because of how much it tosses transmutes across the codebase. Since fn-to-different-fn isn't a primitive cast, `as` is not legal, and transmute is what I end up having to use :(

Reply:
Hopefully a future PR will replace this with an `Instance::run_from_hostcall()` function that will both obviate the need for the user to care about `FunctionHandles` and typecheck the function rather than using `transmute`.